### DISCUSSION TITLE
Rename nodesets

### DISCUSSION THREAD

**Discussion by TheGreatCid** [idaholab/moose#26042] [Q&A General] [answered]
Hi all,

I am using `GeneratedMeshGenerator` to create a 3D mesh. I want the names of the generated sidesets to differ from their corresponding side sets. 
The only way I can see to do this is by renaming all the node sets using `BoundingBoxNodeSetGenerator`, but that is a tedious solution. Is there a straightforward way to do this?

Thanks!

**Comment by lindsayad** [comment, marked as answer]
Yea I don't have a great solution for you. @roystgnr we might consider adding an option to create different IDs for the sidesets/nodesets in `build_cube`?

**Reply by GiudGiud** [reply]
you can use the ParsedGenerateSideset with the restriction options to fairly easily rename sidesets.
If you prefer renaming the sideset, feel free to modify ParsedGenerateSideset into a ParsedGenerateNodeset (make a new object that is) and we're happy to see this in the framework

**Reply by TheGreatCid** [reply]
Got it, thanks!

**Reply by roystgnr** [reply]
We have writable `BoundaryInfo::sideset_name()`, `BoundaryInfo::nodeset_name()`, etc. accessors.  In general I'm in favor of requiring user code to make n calls to `foo(); bar(); baz();` rather than adding 2ⁿ API options for `foo_bar_baz();` vs `foo_baz();` vs ... etc., especially in cases like this where the optional calls are O(1).

For user input decks, on the other hand, there's something to be said for more brevity.  A `GeneratedMeshGenerator` option to override the default naming might be a good idea.
